lib/resources/dstu2/medication_statement.yaml

```diff
@@ -0,0 +1,287 @@
+      - display: SNOMED CT
+        system: http://snomed.info/sct
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.asNeeded_x_
+
+  - name: route
+    required: 'No'
+    cardinality: 0..1
+    type: CodeableConcept
+    description: >
+                     A code specifying the route or physiological path of administration of a therapeutic
+                     agent into or onto a subject.
+    example: |
+              {"route": {
+                "coding": [
+                  {"system": "http://snomed.info/sct",
+                   "code": "41974700",
+                   "display": "Chew - dosing instruction imperative (qualifier value)"}],
+                "text": "Chewed"}}
+    binding:
+      description: >
+                      A coded concept describing the route or physiological path of administration of a
+                      therapeutic agent into or onto the body of a subject.
+      terminology:
+      - display: SNOMED CT
+        system: http://snomed.info/sct
+
+  - name: quantity[x]
+    required: 'No'
+    cardinality: 0..1
+    type: SimpleQuantity
+    description: The amount of therapeutic or other substance given at one administration event.
+    example: |
+               {"quantityQuantity": {
+                 "value": 2,
+                 "unit": "tabs",
+                 "system": "http://unitsofmeasure.org",
+                 "code": "tbl"}}
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.quantity_x_
+
+- name: patient
+  required: 'Yes'
+  cardinality: 1..1
+  type: Reference (Patient)
+  description: Who is/was taking the medication.
+  example: '{"patient": {"reference": "Patient/5366327"}}'
+
+- name: status
+  required: 'Yes'
+  cardinality: 1..1
+  type: code
+  description: >
+                 A code representing the patient or other source's judgment about the state of the medication
+                 used that this statement is about.
+  example: '{"status": "active"}'
+
+- name: note
+  required: 'No'
+  cardinality: 0..1
+  type: string
+  description: Provides extra information about the medication statement that is not conveyed by the other attributes.
+  example: '{"note": "do not take with alcohol"}'
+  action: create
+
+- name: effective[x]
+  required: 'No'
+  cardinality: 0..1
+  type: Period
+  description: The interval of time during which it is being asserted that the patient was taking the medication.
+  note: If effectivePeriod and dosage.timing.repeat.boundsPeriod are both populated they need to be the same value.
+  example: |
+             {"effectivePeriod": {
+               "start": "2015-05-05T14:00:00.000Z",
+               "end": "2015-05-06T01:00:00.000Z"}}
+  url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.effective_x_
+  action: create
+
+- name: medication[x]
+  required: 'Yes'
+  cardinality: 1..1
+  type: Reference (Medication)
```

Comment:
Several things for this field:

1. medication[x] -> medicationReference
2. In AllergyIntolerance when describing the type of a contained reference, I actually listed it as a "contained Reference(\<types go here\>)" [\[github\]](https://github.com/cerner/fhir.cerner.com/blob/master/lib/resources/dstu2/allergy_intolerance.yaml#L89) [\[fhir.cerner.com\]](http://fhir.cerner.com/dstu2/allergy-intolerance/#create-reporter) If you do that you can remove 'It must reference a contained medication' from the note
3. The note is update action specific so I would make it an action specific note so that it doesn't render in the create table*
4. reference and display lines in the example should both be indented on more space. Essentially when determining where to indent a child field, align the opening " of the child with the opening " of the parent and then add 2 spaces to the beginning of the line

```
- name: medicationReference
  required: 'Yes'
  cardinality: 1..1
  type: contained Reference (Medication)
  description: Identifies the medication being administered.
  note:  
  - update: This must be the current medicationReference.code. It cannot be modified.
  example: |
             {"medicationReference": {
                "reference": "#456235",
                "display": "Ibuprofen"}}
  url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.medication_x_
```

* It looks like I inadvertently introduced a workflow when adding terminology_table which broke this functionality. It can be fixed by adding an extra check in definition_table_generator.rb. Here's a gist showing the fix: https://gist.github.com/cerbs/582f1b7eae0881aa5708